File: Csharp/Command/Command/Invoker.cs
using System;

namespace Command
{
    internal class Invoker
    {
        private ICommand _onStart;

        private ICommand _onFinish;

        // Initialize commands.
        public void SetOnStart(ICommand command)
        {
            this._onStart = command;
        }

        public void SetOnFinish(ICommand command)
        {
            this._onFinish = command;
        }
        
        public void DoSomethingImportant()
        {
            Console.WriteLine("Invoker: Make a request");
            _onStart?.Execute();

            Console.WriteLine("Invoker: ...doing something really important...");
            
            Console.WriteLine("Invoker: Does anybody want something done after I finish?");
            _onFinish?.Execute();
        }

    }
}

File: PHP/show/Stage/Invoker.php
<?php

class Invoker
{
    private $onStart;
    private $onFinish;

    public function setOnStart(Command $command): void
    {
        $this->onStart = $command;
    }

    public function setOnFinish(Command $command): void
    {
        $this->onFinish = $command;
    }
    public function doSomethingImportant(): void
    {
        echo "Invoker:  Makes a request<br/>";
        if ($this->onStart instanceof Command) {
            $this->onStart->execute();
        }

        echo "Invoker: does the action...<br/>";
        echo "Invoker: Makes another request<br/>";
        if ($this->onFinish instanceof Command) {
            $this->onFinish->execute();
        }
    }
}